### ISSUE TITLE
SDS performance impact / regression

### ISSUE THREAD

**Issue by costinm** [istio/istio#15450]
Epic: https://github.com/istio/istio/issues/15267

When SDS is enabled, EVERY cluster and listeners gets an additional large config (sample at end). This results in GC and other overhead - and appears to have an impact on Pilot CPU usage and likely on scalability. 

For beta we must at least have a measurement, so CPU allocs can be increased - and if the impact is too big we need to optimize this a bit.

For example the root CAs do not need SDS or tokens - can continue to use mounted files ( and share the root CA rotation / migration  / multicluster work that continues to be done ). 

Since the config is identical in all clusters - it is not clear why we're not using a single Envoy cluster and avoid all this overhead. It might be due to the token, or other reasons - it would be useful to find out. 

```
{  
          "filterChains": [
            {
                "filterChainMatch": {
                    "applicationProtocols": [
                        "istio"
                    ]
                },
                "tlsContext": {
                    "commonTlsContext": {
                        "tlsCertificateSdsSecretConfigs": [
                            {
                                "name": "default",
                                "sdsConfig": {
                                    "apiConfigSource": {
                                        "apiType": "GRPC",
                                        "grpcServices": [
                                            {
                                                "googleGrpc": {
                                                    "targetUri": "/var/run/sds",
                                                    "channelCredentials": {
                                                        "localCredentials": {}
                                                    },
                                                    "callCredentials": [
                                                        {
                                                            "fromPlugin": {
                                                                "name": "envoy.grpc_credentials.file_based_metadata",
                                                                "typedConfig": {
                                                                    "@type": "type.googleapis.com/envoy.config.grpc_credential.v2alpha.FileBasedMetadataConfig",
                                                                    "secretData": {
                                                                        "filename": "/var/run/secrets/kubernetes.io/serviceaccount/token"
                                                                    },
                                                                    "headerKey": "istio_sds_credentials_header-bin"
                                                                }
                                                            }
                                                        }
                                                    ],
                                                    "statPrefix": "sdsstat",
                                                    "credentialsFactoryName": "envoy.grpc_credentials.file_based_metadata"
                                                }
                                            }
                                        ]
                                    },
                                    "initialFetchTimeout": "0s"
                                }
                            }
                        ],
                        "combinedValidationContext": {
                            "defaultValidationContext": {},

                            "validationContextSdsSecretConfig": {
                                "name": "ROOTCA",
                                "sdsConfig": {
                                    "apiConfigSource": {
                                        "apiType": "GRPC",
                                        "grpcServices": [
                                            {
                                                "googleGrpc": {
                                                    "targetUri": "/var/run/sds",
                                                    "channelCredentials": {
                                                        "localCredentials": {}
                                                    },
                                                    "callCredentials": [
                                                        {
                                                            "fromPlugin": {
                                                                "name": "envoy.grpc_credentials.file_based_metadata",
                                                                "typedConfig": {
                                                                    "@type": "type.googleapis.com/envoy.config.grpc_credential.v2alpha.FileBasedMetadataConfig",
                                                                    "secretData": {
                                                                        "filename": "/var/run/secrets/kubernetes.io/serviceaccount/token"
                                                                    },
                                                                    "headerKey": "istio_sds_credentials_header-bin"
                                                                }
                                                            }
                                                        }
                                                    ],
                                                    "statPrefix": "sdsstat",
                                                    "credentialsFactoryName": "envoy.grpc_credentials.file_based_metadata"
                                                }
                                            }
                                        ]
                                    },
                                    "initialFetchTimeout": "0s"
                                }
                            }
                        },
```

**Describe the feature request**

**Describe alternatives you've considered**

**Affected product area (please put an X in all that apply)**

[ ] Configuration Infrastructure
[ ] Docs
[ ] Installation
[X] Networking
[ X] Performance and Scalability
[ ] Policies and Telemetry
[ X] Security
[ ] Test and Release
[ ] User Experience
[ ] Developer Infrastructure

**Additional context**

**Comment by costinm** [comment]
To be clear: this config is larger than what we typically generate for listener/cluster - it more than doubles the size of each object, and in some cases we still have N**2 config objects in the mesh. It's not a single config object, but one per service, sent to every workload.

**Comment by myidpt** [comment]
Jimmy, can we put this into our plans?

**Comment by howardjohn** [comment]
I think this is fixed by #21108 as we share a cluster for everything now
